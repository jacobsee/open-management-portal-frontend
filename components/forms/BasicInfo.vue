<template>
  <form novalidate class="pf-c-form">
    <div class="pf-c-form__group">
      <label class="pf-c-form__label" for="customer-name">
        <span class="pf-c-form__label-text">Customer</span>
        <span class="pf-c-form__label-required" aria-hidden="true">
          &#42;
        </span>
      </label>
      <input
        class="pf-c-form-control"
        required
        type="text"
        v-model="customer"
        @change="$emit('update:customer', customer)"
      />
      <p class="pf-c-form__helper-text" aria-live="polite">
        The customer name to use for this residency
      </p>
    </div>
    <div class="pf-c-form__group">
      <label class="pf-c-form__label" for="customer-location">
        <span class="pf-c-form__label-text">Location</span>
        <span class="pf-c-form__label-required" aria-hidden="true">
          &#42;
        </span>
      </label>
      <input
        class="pf-c-form-control"
        required
        type="email"
        v-model="location"
        @change="$emit('update:location', location)"
      />
      <p class="pf-c-form__helper-text" aria-live="polite">
        Where the residency will be held
      </p>
    </div>
    <div class="pf-c-form__group">
      <label class="pf-c-form__label" for="simple-form-number">
        <span class="pf-c-form__label-text">Dates</span>
      </label>
      <br />
      <template>
        <client-only>
          <date-picker
            v-model="dates"
            input-class="pf-c-form-control"
            range
            @change="$emit('update:dates', dates)"
          />
        </client-only>
      </template>
      <p class="pf-c-form__helper-text" aria-live="polite">
        When the residency will be held
      </p>
    </div>
  </form>
</template>

<script>
export default {
  props: ['customer', 'location', 'dates']
}
</script>

<style>
.pf-c-form-control {
  width: 100%;
}
.mx-datepicker {
  width: 100%;
}
.mx-input-wrapper {
  width: 100%;
}
td.cell {
  text-align: center;
}
</style>
