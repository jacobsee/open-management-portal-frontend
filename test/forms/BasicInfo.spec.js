import { shallowMount, createLocalVue } from '@vue/test-utils'
import DatePicker from 'vue2-datepicker'
import BasicInfo from '@/components/forms/BasicInfo.vue'

const localVue = createLocalVue()
localVue.component('date-picker', DatePicker)

const testProps = {
  customer: 'me',
  location: 'somewhere',
  dates: [
    Date.parse('01 Jan 1970 00:00:00 GMT'),
    Date.parse('04 Dec 1995 00:12:00 GMT')
  ]
}

const TestDate = Date

beforeEach(() => {
  global.Date = class extends TestDate {
    constructor() {
      super()
      return new TestDate('01 Jan 1970 00:00:00 GMT')
    }
  }
})

afterEach(() => {
  global.Date = TestDate
})

describe('BasicInfo', () => {
  test('is a Vue instance', () => {
    const wrapper = shallowMount(BasicInfo, {
      propsData: testProps,
      localVue
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders properly', () => {
    const wrapper = shallowMount(BasicInfo, {
      propsData: testProps,
      localVue
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
