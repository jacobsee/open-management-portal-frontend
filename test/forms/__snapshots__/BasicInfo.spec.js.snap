// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInfo renders properly 1`] = `
"<form novalidate=\\"novalidate\\" class=\\"pf-c-form\\"><div class=\\"pf-c-form__group\\"><label for=\\"customer-name\\" class=\\"pf-c-form__label\\"><span class=\\"pf-c-form__label-text\\">Customer</span> <span aria-hidden=\\"true\\" class=\\"pf-c-form__label-required\\">
        *
      </span></label> <input required=\\"required\\" type=\\"text\\" class=\\"pf-c-form-control\\"> <p aria-live=\\"polite\\" class=\\"pf-c-form__helper-text\\">
      The customer name to use for this residency
    </p></div> <div class=\\"pf-c-form__group\\"><label for=\\"customer-location\\" class=\\"pf-c-form__label\\"><span class=\\"pf-c-form__label-text\\">Location</span> <span aria-hidden=\\"true\\" class=\\"pf-c-form__label-required\\">
        *
      </span></label> <input required=\\"required\\" type=\\"email\\" class=\\"pf-c-form-control\\"> <p aria-live=\\"polite\\" class=\\"pf-c-form__helper-text\\">
      Where the residency will be held
    </p></div> <div class=\\"pf-c-form__group\\"><label for=\\"simple-form-number\\" class=\\"pf-c-form__label\\"><span class=\\"pf-c-form__label-text\\">Dates</span></label> <br> <client-only-stub placeholdertag=\\"div\\"><date-picker-stub value=\\"0,818035920000\\" defaultvalue=\\"Tue Nov 26 2019 00:00:00 GMT-0800 (Pacific Standard Time)\\" disableddate=\\"function _default() {
          return false;
        }\\" type=\\"date\\" getclasses=\\"function _default() {
          return [];
        }\\" titleformat=\\"YYYY-MM-DD\\" format=\\"YYYY-MM-DD\\" timetitleformat=\\"YYYY-MM-DD\\" disabledtime=\\"function _default() {
          return false;
        }\\" hourstep=\\"1\\" minutestep=\\"1\\" secondstep=\\"1\\" valuetype=\\"date\\" range=\\"true\\" rangeseparator=\\" ~ \\" placeholder=\\"\\" editable=\\"true\\" clearable=\\"true\\" inputclass=\\"pf-c-form-control\\" inputattr=\\"[object Object]\\" appendtobody=\\"true\\" popupstyle=\\"[object Object]\\" confirmtext=\\"OK\\" shortcuts=\\"\\"></date-picker-stub></client-only-stub> <p aria-live=\\"polite\\" class=\\"pf-c-form__helper-text\\">
      When the residency will be held
    </p></div></form>"
`;
